lib/watir/window.rb

```diff
@@ -3,8 +3,14 @@ class Window
     include EventuallyPresent
     include Waitable
 
-    def initialize(driver, selector)
-      @driver = driver
+    def initialize(browser, selector)
+      if browser.is_a?(Watir::Browser)
+        @browser = browser
+        @driver = browser.driver
+      else
```

Comment:
currently this method is initialized by a driver, not a browser. Most likely no one initializes a window directly and goes through Browser, but I don't think this is a private API. which is why I put the deprecation notice.